homeassistant/components/reolink/binary_sensor.py

```diff
@@ -0,0 +1,164 @@
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, PERSON_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, PERSON_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key=VEHICLE_DETECTION_TYPE,
+        name=VEHICLE_DETECTION_TYPE,
+        icon="mdi:car",
+        icon_off="mdi:car-off",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, VEHICLE_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, VEHICLE_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key=PET_DETECTION_TYPE,
+        name=PET_DETECTION_TYPE,
+        icon="mdi:dog-side",
+        icon_off="mdi:dog-side-off",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, PET_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, PET_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key="visitor",
+        name="Visitor",
+        icon="mdi:bell-ring-outline",
+        icon_off="mdi:doorbell",
+        value=lambda host, ch: host.api.visitor_detected(ch),
+        supported=lambda host, ch: host.api.is_doorbell_enabled(ch),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Reolink IP Camera."""
+    reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
+    host = reolink_data.host
+
+    entities = []
+    for channel in host.api.channels:
+        for description in BINARY_SENSORS:
+            if not description.supported(host, channel):
+                continue
+
+            entities.append(
+                ReolinkBinarySensorEntity(
+                    reolink_data,
+                    channel,
+                    description,
+                )
+            )
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class ReolinkBinarySensorEntity(ReolinkCoordinatorEntity, BinarySensorEntity):
+    """An implementation of a base binary-sensor class for Reolink IP camera motion sensors."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        reolink_data: ReolinkData,
+        channel: int,
+        description: ReolinkBinarySensorDescription,
+    ) -> None:
+        """Initialize Reolink binary sensor."""
+        ReolinkCoordinatorEntity.__init__(self, reolink_data, channel)
+        BinarySensorEntity.__init__(self)
+
+        self._description = description
+
+        self._attr_name = description.name
+        self._attr_unique_id = (
+            f"{self._host.unique_id}_{self._channel}_{description.key}"
+        )
+        self._attr_device_class = description.device_class
+
+    @property
+    def icon(self) -> str:
+        """Icon of the sensor."""
+        if self.is_on:
```

Comment:
It would be better to check the internal property as this implementation is relying on the order in which `is_on` is called in the parent

Reply:
done